src/js/actions/index.js

```diff
@@ -485,22 +499,14 @@ export function fetchNewSubmission(id, period) {
  */
 export function fetchSubmission() {
   return dispatch => {
-    dispatch(requestFiling())
-    return getFilingFromUrl().then(json => {
-        dispatch(receiveFiling(json))
-
-        const latestSubmission = json.submissions.reduce((prev, curr) => {
-            return +curr.id.sequenceNumber > +prev.id.sequenceNumber ? curr : prev
-          }, {id: {sequenceNumber: 0}})
-
-        if(latestSubmission.id.sequenceNumber !== 0){
-          return dispatch(receiveSubmission(latestSubmission))
+    return getLatestSubmission()
+      .then(json => {
+        if(json.httpStatus === 404){
+          const splitPath = json.path.split('/')
+          return dispatch(fetchNewSubmission(splitPath[2], splitPath[4]))
         }else{
-          return createSubmission(json.filing.institutionId, json.filing.period).then(submission => {
-              dispatch(receiveSubmission(submission))
-            })
+          return setTimeout(()=>dispatch(receiveSubmission(json)),1000)
```

Comment:
Only for demo purposes. I'll pull this out before merge.

The wait was actually quite helpful in discovering and fixing some latent race conditions